daemon/src/send_future_ext.rs

```diff
@@ -0,0 +1,35 @@
+use async_trait::async_trait;
+use std::error::Error;
+
+#[async_trait]
+pub trait Warn {
+    async fn warn(self);
+}
+
+#[async_trait]
+impl<E: Error + Send, A: xtra::Actor, M: xtra::Message<Result = Result<(), E>>> Warn
+    for xtra::address::SendFuture<A, M>
+{
+    async fn warn(self) {
+        let msg = match self.await {
+            Err(e) => format!("{:#}", e),
```

Comment:
I don't think we've been making a clear distinction in terms of handling actor disconnects and handler failures. Doing this would suggest to me that we should almost always bubble up actor disconnects. Is that what you see?

Otherwise, wouldn't this change just make this trait method less convenient?

Reply:
We should be making a distinction as in most cases, an actor being disconnected is a bug I'd say? Now the question is, do we want to handle that in one place or explicitely at every call site?

The only example I can think of where it might not be a bug is if we tie the lifetime of a network connection to an actor.

Reply:
I agree that it's always unexpected when the receiving actor is down. But I think we have been primarily logging these errors, so I suspect if we return the `Disconnected` error we're just gonna log at every call site.

If you want the flexibility of being able to handle the `Disconnected` error, could we introduce another trait which warns about the disconnect. I'm just worried about ergonomics and code readability if we just match against the error everywhere.